tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp

```diff
@@ -21,20 +21,56 @@ struct LayoutMapper<std::experimental::layout_right> {
   using type = Kokkos::LayoutRight;
 };
 
+template<class T>
+constexpr void static_assert_complex_alignment(const T* /* p */)
+{
+  using kc_t = Kokkos::complex<typename T::value_type>;
+  static_assert(alignof(T) == alignof(kc_t),
+		"KokkosBlas: mdspan_to_view: mismatching alignment of std::complex<> and Kokkos::complex<> which prevents interoperability. By default, Kokkos::complex<T> is aligned as 2*sizeof(T) but std::complex<T> is aligned as sizeof(T). To match the alignment and allow correct conversion, you must rebuild Kokkos with KOKKOS_ENABLE_COMPLEX_ALIGN undefined.");
+}
+
+//
+// to_kokkos_pointer
+//
+template<class ValueType>
+ValueType* to_kokkos_pointer(ValueType* p) {
+  return p;
+}
+
+template<class RealType>
+auto to_kokkos_pointer(std::complex<RealType>* p) {
+  static_assert_complex_alignment(p);
+  return reinterpret_cast<Kokkos::complex<RealType>*>(p);
+}
+
+template<class RealType>
+auto to_kokkos_pointer(const std::complex<RealType>* p) {
+  static_assert_complex_alignment(p);
+  return reinterpret_cast<const Kokkos::complex<RealType>*>(p);
+}
+
+//
+// mdspan_to_view
+//
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
-mdspan_to_view(const MDSpan& a) {
-    return Kokkos::View<typename MDSpan::element_type*>(a.data(),a.extent(0));
+requires (MDSpan::rank()==1)
+auto mdspan_to_view(const MDSpan& a) {
```

Comment:
Yes good point, I changed it to be explicitly a mdspan. I totally agree that if make this generic, the name should be changed. So since we keep the name for now, I changed it so that it actually reflects that.